metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/EnrichmentPlugin.java

```diff
@@ -8,12 +8,20 @@
  */
 public class EnrichmentPlugin extends AbstractMetisPlugin {
 
+  /**
+   * Zero argument constructor that initializes the {@link #pluginType} corresponding to the plugin.
+   */
   public EnrichmentPlugin() {
-    super();
-    setPluginType(PluginType.ENRICHMENT);
     //Required for json serialization
+    super.setPluginType(PluginType.ENRICHMENT);
```

Comment:
Why call super? Maybe later we need to override this method, and we may miss this call. I think it is generally best not to use super unless we have to.

This also holds for all sibling classes.

Reply:
You are completely right. The reason I had to do this is because I need to initialize on the constructor and so the user would not forget to use the "setter" method.
The other option is too actually remove the pluginType field to the subclasses and make the setter on the abstract class abstract so that full control will be on the subclasses. What do you think?

Reply:
I think what I was proposing is to write 
setPluginType(PluginType.ENRICHMENT);
instead of 
super.setPluginType(PluginType.ENRICHMENT);
You can even make setPluginType final as a method.

But now that I look at it more, I think the plugin type should never be allowed to change, so at least it should be final (as a property of the superclass): only to be set on construction.

Alternatively you could remove the property of the superclass and just make an abstract getter for the plugin type that the subclasses are then forced to implement.

Reply:
Ok your last point was my initial thought as well. I didn't remember but I just tried it and I saw the error again and remembered that morphia gives me if I move the pluginType inside the implemented class. So basically the pluginType has to be part of the AbstractMetisPlugin because morphia requires it when I am doing a search.

Error:
The field 'pluginType' could not be found in 'eu.europeana.metis.core.workflow.plugins.AbstractMetisPlugin' while validating - metisPlugins.pluginType; if you wish to continue please disable validation.

If now I would have the pluginType in AbstractMetisPlugin and shadow it in the implemented class that also doesn't work with morphia because of error:
Mapping to MongoDB field name 'pluginType' is duplicated; you cannot map different java fields to the same MongoDB field.

At the end I ended up on a different way to handle this, which I think it's the best possible in this situation.